helix-term/src/commands/typed.rs

```diff
@@ -1943,73 +1829,94 @@ fn toggle_option(
     if args.is_empty() {
         anyhow::bail!("Bad arguments. Usage: `:toggle key [values]?`");
     }
-    let key = &args[0].to_lowercase();
+    let key = args.next().unwrap().to_lowercase();
 
-    let key_error = || anyhow::anyhow!("Unknown key `{}`", key);
-
-    let mut config = serde_json::json!(&cx.editor.config().deref());
+    let mut config = serde_json::json!(&*cx.editor.config());
     let pointer = format!("/{}", key.replace('.', "/"));
-    let value = config.pointer_mut(&pointer).ok_or_else(key_error)?;
+    let value = config
+        .pointer_mut(&pointer)
+        .ok_or_else(|| anyhow::anyhow!("Unknown key `{}`", key))?;
 
     *value = match value {
         Value::Bool(ref value) => {
             ensure!(
-                args.len() == 1,
+                args.next().is_none(),
                 "Bad arguments. For boolean configurations use: `:toggle key`"
             );
             Value::Bool(!value)
         }
         Value::String(ref value) => {
             ensure!(
-                args.len() > 2,
+                args.count() >= 2,
```

Comment:
this will cause issues I think, did you try this out?

Iterator::count will advance the iterator so afterwards all arguments will be consumed

Reply:
The reason I got tripped up here is that iterators are never supposed to be copy in rust
(official recommendations by the project). So I would definitly remove that.

I think `arg_count(&self)` makes sese

Reply:
Decided to forgo the count use as talked about. Just used a `fold` to count during creation. If count was used even inside the wrapper it would take from the current state, which would be wrong as `arg_count` is the absolute count. Even if I cloned inside it would clone the potentially already progressed state. Add a `count` field to hold this state.

The panic message isn't strictly needed, but I think the correct call would be to `arg_count` in 99% of use cases. Any use that could be needed from the "relative" count should be covered by just subtracting when needed. And if there is a usecase that really really needs it than they can just remove the impl for it and we can go from there.